extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -260,18 +283,40 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 }
 
 // RestoreRKE2K3SSnapshot is a helper function to restore a snapshot on an RKE2 or k3s cluster. Returns error if any.
-func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, initialControlPlaneValue, initialWorkerValue string) error {
-	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
+func RestoreRKE2K3SSnapshot(client *rancher.Client, snapshotRestore *rkev1.ETCDSnapshotRestore, cluster *apisV1.Cluster) error {
+	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, cluster.Name, fleetNamespace)
 	if err != nil {
 		return err
 	}
 
 	clusterObject.Spec.RKEConfig.ETCDSnapshotRestore = snapshotRestore
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = initialControlPlaneValue
-	clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = initialWorkerValue
+	clusterObject.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency = cluster.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency
+	clusterObject.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency = cluster.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency
```

Comment:
same comment here on having access to something that rancher doesn't.

Reply:
Discussed offline, resolved.